srtcore/buffer_rcv.cpp

```diff
@@ -254,53 +254,51 @@ int CRcvBuffer::dropAll()
     return dropUpTo(end_seqno);
 }
 
-int CRcvBuffer::dropMessage(int32_t seqnolo, int32_t seqnohi, int32_t msgno)
+int CRcvBuffer::dropMessage(int32_t seqnolo, int32_t seqnohi, int32_t msgno, DropActionIfExists actionOnExisting)
 {
     IF_RCVBUF_DEBUG(ScopedLog scoped_log);
     IF_RCVBUF_DEBUG(scoped_log.ss << "CRcvBuffer::dropMessage: seqnolo " << seqnolo << " seqnohi " << seqnohi << " m_iStartSeqNo " << m_iStartSeqNo);
+    const bool bKeepExisting = (actionOnExisting == KEEP_EXISTING);
     // TODO: count bytes as removed?
     const int end_pos = incPos(m_iStartPos, m_iMaxPosOff);
+    int minDroppedOffset = -1;
+    int iDropCnt = 0;
     if (msgno > 0) // including SRT_MSGNO_NONE and SRT_MSGNO_CONTROL
     {
+        // First try to drop by message number in case the message starts earlier thtan @a seqnolo.
         IF_RCVBUF_DEBUG(scoped_log.ss << " msgno " << msgno);
-        int minDroppedOffset = -1;
-        int iDropCnt = 0;
         for (int i = m_iStartPos; i != end_pos; i = incPos(i))
         {
-            // TODO: Maybe check status?
+            // Can't drop is message number is not known.
+            // TODO: Maybe check entry status?
             if (!m_entries[i].pUnit)
                 continue;
 
-            // TODO: Break the loop if a massege has been found. No need to search further.
             const int32_t msgseq = packetAt(i).getMsgSeq(m_bPeerRexmitFlag);
             if (msgseq == msgno)
```

Comment:
Note that message numbers are also increasing with rollover. It might be a good idea to check if the message number from the current packet is already PAST the searched message number. If so, there's no point in continuing with the loop because remaining packets in the buffer won't contain this message.